keras_benchmark/scripts/benchmark_resnet.py

```diff
@@ -0,0 +1,353 @@
+
+    # Normalize data.
+    x_train = x_train.astype('float32') / 255
+    x_test = x_test.astype('float32') / 255
+
+    # If subtract pixel mean is enabled
+    if subtract_pixel_mean:
+        x_train_mean = np.mean(x_train, axis=0)
+        x_train -= x_train_mean
+        x_test -= x_train_mean
+    #x_train = x_train[:200]
+    #y_train = y_train[:200]
+    #x_test = x_test[:200]
+    #y_test = y_test[:200]
+    print('x_train shape:', x_train.shape)
+    print(x_train.shape[0], 'train samples')
+    print(x_test.shape[0], 'test samples')
+    print('y_train shape:', y_train.shape)
+
+    # Convert class vectors to binary class matrices.
+    y_train = keras.utils.to_categorical(y_train, num_classes)
+    y_test = keras.utils.to_categorical(y_test, num_classes)
+
+# ImageNet Dataset
+if args.dataset == "imagenet":
+    input_shape = (256, 256, 3) if data_format == 'channels_last' else (3, 256, 256)
+    if args.train_mode == 'fit_generator':
+        train_datagen = ImageDataGenerator(
+            rescale=1. / 255)
+
+        test_datagen = ImageDataGenerator(rescale=1. / 255)
+
+        train_generator = train_datagen.flow_from_directory(
+            args.data_path,
+            target_size=(256, 256),
+            batch_size=batch_size)
+    else:
+        # Load the imagenet data.
+        train_images = []
+        train_labels = []
+        label_counter = 0
+        for subdir, dirs, files in os.walk(args.data_path):
+            for folder in dirs:
+                for folder_subdir, folder_dirs, folder_files in \
+                        os.walk(os.path.join(subdir, folder)):
+                    for file in folder_files:
+                        train_images.append(os.path.join(folder_subdir, file))
+                        train_labels.append(label_counter)
+
+                label_counter = label_counter + 1
+
+        # shuffle data
+        perm = list(range(len(train_images)))
+        random.shuffle(perm)
+        train_images = [train_images[index] for index in perm]
+        train_labels = [train_labels[index] for index in perm]
+        nice_n = math.floor(len(train_images) / batch_size) * batch_size
+
+
+# process batch data for imagenet
+def get_batch():
+    index = 1
+
+    global current_index
+
+    B = np.zeros(shape=(batch_size, input_shape[0], input_shape[1], input_shape[2]))
+    L = np.zeros(shape=(batch_size))
+
+    while index < batch_size:
+        try:
+            img = load_img(train_images[current_index].rstrip(),
+                           target_size=(256, 256, 3))
+            B[index] = img_to_array(img)
+            B[index] /= 255
+
+            L[index] = train_labels[current_index]
+            index = index + 1
+            current_index = current_index + 1
+        except:
+            print("Ignore image {}".format(train_images[current_index]))
+            current_index = current_index + 1
+
+    return B, keras.utils.to_categorical(L, num_classes)
+
+
+# Prepare Model
```

Comment:
move this to README file - any reason why this has been included with the code?

Reply:
These results are from official TensorFlow benchmark repo and the data is from original papers and not from our benchmark script. So, didn't move to the README file.